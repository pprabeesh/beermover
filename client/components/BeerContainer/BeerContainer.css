.item {
  margin: 10px;
  color: #151515;
  min-height: 20rem;
  min-width: 20rem;
  background-color: #74a57f;
  font-size: larger;
  text-align: center;
  display: flex;
  align-items: center;
  justify-content: center;
  flex-direction: column;
}

.data {
  margin: 0;
  padding: 2%;
  display: inline-block;
  width: 100%;
}

.notify {
  background-color: hotpink;
}
