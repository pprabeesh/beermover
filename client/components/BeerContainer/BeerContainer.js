import React from "react";
import "./BeerContainer.css";

const BeerContainer = props => {
  const { container, currentTemps } = props;
  const containerData = currentTemps.find(
    currentTemp => currentTemp.id === container.name
  );

  let notify = false;
  let temperature = "";
  if (containerData) {
    temperature = containerData.temperature;
    notify = temperature > container.maxTemp || temperature < container.minTemp;
  }
  return (
    <div className={` item ${notify ? "notify" : ""}`}>
      <div className="data"> {container.name}</div>
      <div className="data">
        {temperature || temperature === 0 ? `${temperature}Â°C` : ""}
      </div>
    </div>
  );
};

export default BeerContainer;
