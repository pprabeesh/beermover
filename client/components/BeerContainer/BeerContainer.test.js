import React from "react";
import { mount, shallow } from "enzyme";
import BeerContainer from "./BeerContainer";

const props = {
  container: {
    name: "testContainer1",
    maxTemp: 4,
    minTemp: 3
  },
  currentTemps: [
    {
      id: "testContainer1",
      temperature: 3
    }
  ]
};

describe("BeerContainer", () => {
  it("renders snapshots", () => {
    const wrapper = shallow(<BeerContainer {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it(" renders with class", () => {
    const wrapper = mount(<BeerContainer {...props} />);
    expect(wrapper.find("div")).toHaveLength(3);
    expect(wrapper.find(".item")).toHaveLength(1);
    expect(wrapper.find(".data")).toHaveLength(2);
    expect(wrapper.find(".notify")).toHaveLength(0);
  });

  it(" renders with notify clas", () => {
    props.currentTemps[0].temperature = 5;
    const wrapper = mount(<BeerContainer {...props} />);
    expect(wrapper.find("div")).toHaveLength(3);
    expect(wrapper.find(".item")).toHaveLength(1);
    expect(wrapper.find(".data")).toHaveLength(2);
    expect(wrapper.find(".notify")).toHaveLength(1);
  });

  it(" renders without current temp data", () => {
    props.currentTemps = [];
    const wrapper = mount(<BeerContainer {...props} />);
    expect(wrapper.find("div")).toHaveLength(3);
    expect(wrapper.find(".item")).toHaveLength(1);
    expect(wrapper.find(".data")).toHaveLength(2);
    expect(wrapper.find(".notify")).toHaveLength(0);
  });
});
