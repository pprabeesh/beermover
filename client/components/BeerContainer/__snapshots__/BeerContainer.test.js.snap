// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BeerContainer renders snapshots 1`] = `ShallowWrapper {}`;
